rust/lsp-lib/src/conversion_utils.rs

```diff
@@ -116,3 +126,47 @@ pub(crate) fn core_hover_from_hover<C: Cache>(
         },
     })
 }
+
+fn position_to_offset(rope: &mut Rope, position: Position) -> usize {
+    let line_utf16_offset =
+        rope.convert_metrics::<LinesMetric, Utf16CodeUnitsMetric>(position.line as usize);
+    let utf16_offset = line_utf16_offset + (position.character as usize);
+    rope.convert_metrics::<Utf16CodeUnitsMetric, BaseMetric>(utf16_offset)
+}
+
+pub(crate) fn core_location_from_location<C: Cache>(plugin_context: &mut PluginContext<C>, location: &Location) -> CoreLocation {
+    let path = location.uri.to_file_path().unwrap();
+    let view = plugin_context.get_view_for_path(&path);
+    
+    let contents = match view {
+        Some(view) => view.get_document().unwrap(),
```

Comment:
It seems unfortunate to always be doing this work when we have the view open; we should be able to just get the line we need and then do the utf16 count on it (and we could still convert to a rope in that case, which is convenient). I don't think this is _super_ important, though.